### PR TITLE
fix: Threshold-precision (improve algebraic stability)

### PR DESCRIPTION
**Pull request by DaveSkender**
Fix rounding issue with bounded indicators like Stochastic and Williams %R by introducing standard precision output, to avoid rounding errors overlapping ranges like 0 → 100 oscillators.

To do:

- [x] Rollback or separate `_testdata` folder rename
- [x] Rollback most of these changes and reconsider only applying where boundary cannot be met with algebraically stable formulations
- [x] Consider more thoughtful test scenarios to find actual out of bounds realistic data — do not assume clamping is needed
- [x] If even needed, only apply `ToPrecision` rounding when actually out of bounds within precision limit — or just set to explicit boundary value.  Throw OutOfBounds kind of exception if exceeds threshold limit of minor rounding

### CODE REVIEW

File: tests/indicators/s-z/WilliamsR/WilliamsR.StaticSeries.Tests.cs [hunk 1]
```diff
@@ -95,30 +95,59 @@ public void Boundary()
...
+
+        Console.WriteLine(results.ToStringOut(args));
+
+        // analyze boundary
+        for (int i = 0; i < length; i++)
+        {
+            Quote q = quotes[i];
+            WilliamsResult r = results[i];
 
             r.WilliamsR?.Should().BeInRange(-100d, 0d);
```
[the remark this answers is not in the document]
Comment (DaveSkender) [reply tests/indicators/s-z/WilliamsR/WilliamsR.StaticSeries.Tests.cs]: I'm going to fix the underlying rounding issue, not fudge the test.

[the remark this answers is not in the document]
Comment (DaveSkender) [reply tests/indicators/s-z/WilliamsR/WilliamsR.StaticSeries.Tests.cs]: Yes, help please, find the right place to apply minimimally sufficient rounding to ensure we stay in bounds of final result.

[the remark this answers is not in the document]
Comment (DaveSkender) [reply tests/indicators/s-z/WilliamsR/WilliamsR.StaticSeries.Tests.cs]: @copilot apply this fix

File: docs/plans/threshold-precision.plan.md
Comment (DaveSkender) [inline docs/plans/threshold-precision.plan.md:1]: Key points from `docs/plans/threshold-precision.plan.md` should go into the instructions, skills, or agent profiles, then deleted.

File: tests/indicators/s-z/Stoch/Stoch.StreamHub.Tests.cs [hunk 2]
```diff
@@ -63,6 +63,16 @@ public override void ToStringOverride_ReturnsExpectedName()
...
 
+    [TestMethod]
+    public void Results_AreAlwaysBounded()
+    {
+        QuoteHub provider = new();
+        StochHub hub = new(provider, 14, 3, 3);
+        provider.Add(Quotes);
+        hub.Results.IsBetween(static x => x.Oscillator, 0d, 100d);
+        hub.Results.IsBetween(static x => x.Signal, 0d, 100d);
+    }
```
[the remark this answers is not in the document]
Comment (DaveSkender) [reply tests/indicators/s-z/Stoch/Stoch.StreamHub.Tests.cs:72]: It’s probably okay to skip the formal terminations here as it’ll auto dispose, right?  These aspects are not subject of test.

File: tests/indicators/s-z/StochRsi/StochRsi.BufferList.Tests.cs [hunk 3]
```diff
@@ -48,6 +48,14 @@ public void QuotesCtor_OnInstantiation_IncrementsResults()
         sut.Should().BeEquivalentTo(series, static options => options.WithStrictOrdering());
     }
 
+    [TestMethod]
+    public void Results_AreAlwaysBounded()
+    {
+        StochRsiList results = new(14, 14, 3, 1, Quotes);
+        results.IsBetween(x => x.StochRsi, 0, 100);
+        results.IsBetween(x => x.Signal, 0, 100);
+    }
```
[the remark this answers is not in the document]
Comment (DaveSkender) [reply tests/indicators/s-z/StochRsi/StochRsi.BufferList.Tests.cs:57]: We’ve made the underlying calcs to be more algebraically stable to fix this.

File: tests/indicators/s-z/StochRsi/StochRsi.StaticSeries.Tests.cs [hunk 4]
```diff
@@ -42,6 +42,16 @@ public override void DefaultParameters_ReturnsExpectedResults()
...
 
+    [TestMethod]
+    public void Results_AreAlwaysBounded()
+    {
+        IReadOnlyList<StochRsiResult> results = Quotes
+            .ToStochRsi(14, 14, 3, 1);
+
+        results.IsBetween(static x => x.StochRsi, 0d, 100d);
+        results.IsBetween(static x => x.Signal, 0d, 100d);
+    }
```
[the remark this answers is not in the document]
Comment (DaveSkender) [reply tests/indicators/s-z/StochRsi/StochRsi.StaticSeries.Tests.cs:51]: We’ve made the underlying calcs to be more algebraically stable to fix this.

File: tests/indicators/s-z/StochRsi/StochRsi.StreamHub.Tests.cs [hunk 5]
```diff
@@ -3,6 +3,15 @@ namespace StreamHub;
...
 public class StochRsiHubTests : StreamHubTestBase, ITestChainObserver, ITestChainProvider
 {
+    [TestMethod]
+    public void Results_AreAlwaysBounded()
+    {
+        StochRsiHub sut = Quotes.ToStochRsiHub(14, 14, 3, 1);
+
+        sut.Results.IsBetween(static x => x.StochRsi, 0d, 100d);
+        sut.Results.IsBetween(static x => x.Signal, 0d, 100d);
+    }
```
[the remark this answers is not in the document]
Comment (DaveSkender) [reply tests/indicators/s-z/StochRsi/StochRsi.StreamHub.Tests.cs:12]: We’ve made the underlying calcs to be more algebraically stable to fix this.
